### DISCUSSION TITLE
IDE feature to refactor redundant null checks

### DISCUSSION THREAD

**Discussion by kandicst** [dotnet/csharplang#7285] [Language Ideas] [answered]
It would maybe make sense in nullable enabled code to find redundant null checks and suggest the user to remove them, or at least have a warning that this will always evaluate to the same value.

```c#
#nullable enable
...

void PrintNotEmpty(string s) 
{
    if (s is null || s == string.Empty)
    {
        return;
    }
  
    Console.Writeline(s);
}
```
Since ```s``` can never be null refactor this to:

```c#
#nullable enable
...

void PrintNotEmpty(string s) 
{
    if (s == string.Empty)
    {
        return;
    }
  
    Console.Writeline(s);
}
```

**Comment by 333fred** [comment]
IDE suggestions should be made over on dotnet/roslyn. This repo is specifically for language design.

**Comment by TahirAhmadov** [comment, marked as answer]
https://github.com/dotnet/csharplang/discussions/4216
